tests/tests/test-xcm/test-xcm-erc20-fees-and-trap.ts

```diff
@@ -0,0 +1,348 @@
+    );
+
+    // Check the error is TooExpensive
+    expect(events).to.have.lengthOf(1);
+    expect(events[0].toHuman().event.data.error).equals("TooExpensive");
+
+    // Charleth should not receive ERC20 tokens due to failed execution
+    expect(
+      (
+        await web3EthCall(context.web3, {
+          to: erc20ContractAddress,
+          data: ERC20_INTERFACE.encodeFunctionData("balanceOf", [CHARLETH_ADDRESS]),
+        })
+      ).result
+    ).equals(bnToHex(0n, { bitLength: 256 }));
+  });
+});
+
+describeDevMoonbeam("Mock XCM - Trap ERC20", (context) => {
+  let erc20Contract: Contract;
+  let erc20ContractAddress: string;
+
+  before("Should deploy erc20 contract", async function () {
+    const { contract, contractAddress } = await setupErc20Contract(context, "NewToken", "NTK");
+    erc20Contract = contract;
+    erc20ContractAddress = contractAddress;
+  });
+
+  it("Should not trap any ERC20", async function () {
+    this.timeout(20_000);
+    const paraId = 888;
+    const paraSovereign = sovereignAccountOfSibling(context, paraId);
+    const amountTransferred = 1_000_000n;
+
+    // Get pallet index
+    const metadata = await context.polkadotApi.rpc.state.getMetadata();
+    const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+      (pallet) => pallet.name === "Balances"
+    ).index;
+    const erc20XcmPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+      (pallet) => pallet.name === "Erc20XcmBridge"
+    ).index;
+
+    // Send some native tokens to the sovereign account of paraId (to pay fees)
+    const { result } = await context.createBlock(
+      createTransaction(context, {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        value: 1_000_000_000_000_000_000,
+        to: paraSovereign,
+        data: "0x",
+      })
+    );
+    expectEVMResult(result.events, "Succeed");
+
+    // Send some erc20 tokens to the sovereign account of paraId
+    const { result: result2 } = await context.createBlock(
+      createTransaction(context, {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        to: erc20ContractAddress,
+        data: ERC20_INTERFACE.encodeFunctionData("transfer", [paraSovereign, amountTransferred]),
+      })
+    );
+    expectEVMResult(result2.events, "Succeed");
+
+    // Check the sovereign account has reveived erc20 tokens
+    expect(
+      (
+        await web3EthCall(context.web3, {
+          to: erc20ContractAddress,
+          data: ERC20_INTERFACE.encodeFunctionData("balanceOf", [paraSovereign]),
+        })
+      ).result
+    ).equals(bnToHex(amountTransferred, { bitLength: 256 }));
+
+    // Create xcm message to send ERC20 tokens to Charleth
+    const config = {
+      assets: [
+        {
+          multilocation: {
+            parents: 0,
+            interior: {
+              X1: { PalletInstance: balancesPalletIndex },
+            },
+          },
+          fungible: 1_000_000_000_000_000n,
```

Comment:
Please use a named constant instead, something like `feeAssetAmount` for instance